cumulus/polkadot-omni-node/lib/src/nodes/manual_seal.rs

```diff
@@ -147,6 +156,17 @@ impl<NodeSpec: NodeSpecT> ManualSealNode<NodeSpec> {
 					.header(block)
 					.expect("Header lookup should succeed")
 					.expect("Header passed in as parent should be present in backend.");
+
+				// If there is a pending validation code in storage, send the go-ahead signal.
+				let should_send_go_ahead = client_for_cidp
+					.storage(block, &StorageKey(PENDING_VALIDATION_CODE_KEY.to_vec()))
+					.ok()
```

Comment:
If someone names `ParachainSystem` differently in `construct_runtime!`, this will fail.

IMO it is better to call `collect_collation_info` at `block` and check if `new_validation_code`  is `Some(_)`.

Reply:
I was totally willing to take that risk :D. But yeah your idea is better. Not checking the runtime api version though, 3 year old runtimes with the manual seal of omni node seems exotic.